input/signdescriptor.go

```diff
@@ -61,6 +61,11 @@ type SignDescriptor struct {
 	// script (PkScript).
 	WitnessScript []byte
 
+	// TaprootKeySpend indicates that instead of a witness script being
+	// spent by the signature that results from this signing request, a
+	// taproot key spend is performed instead.
+	TaprootKeySpend bool
```

Comment:
Assuming we'd add on the additional context here re the leaf being revealed, etc -- I would think that would be enough to go off on (still getting through the latest diff).

Reply:
Ah ok, revisiting the prior commit, I see that we handle the scrip path based on a combo of this bool and the output type. I guess ultimately this single bool is better than exposing 4+ enums that the user need to juggle with.